File: android/src/main/java/com/rebeloid/unity_mediation/initialization/UnityMediationInitialization.java
package com.rebeloid.unity_mediation.initialization;

import com.rebeloid.unity_mediation.UnityMediationConstants;
import com.unity3d.mediation.IInitializationListener;
import com.unity3d.mediation.InitializationConfiguration;
import com.unity3d.mediation.UnityMediation;

import java.util.Map;

import io.flutter.plugin.common.MethodChannel;

public class UnityMediationInitialization {
    private final IInitializationListener initializationListener;

    public UnityMediationInitialization(MethodChannel channel) {
        this.initializationListener = new UnityMediationInitializationListener(channel);
    }

    public boolean initialize(Map<?, ?> arguments) {
        String gameId = (String) arguments.get(UnityMediationConstants.GAME_ID_PARAMETER);
        InitializationConfiguration configuration = InitializationConfiguration.builder()
                .setGameId(gameId)
                .setInitializationListener(initializationListener)
                .build();
        UnityMediation.initialize(configuration);
        return true;
    }
}

File: ios/Classes/UnityMediationInitialization.swift
import UnityMediationSdk

class UnityMediationInitialization {
    let initializationDelegate : UnityMediationInitializationDelegate;
    
    init(channel: FlutterMethodChannel) {
        self.initializationDelegate = UnityMediationInitializationDelegate(channel: channel)
    }
    
    public func initialize(_ args: NSDictionary) -> Bool {
        let gameId = args[UnityMediationConstants.GAME_ID_PARAMETER] as! String
        let configuration = UMSInitializationConfigurationBuilder()
            .setGameId(gameId)
            .setInitializationDelegate(initializationDelegate)
            .build()
        UMSUnityMediation.initialize(with: configuration)
        return true
    }
    
}
